src/contexts/TokenSupply/TokenSupplyProvider.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react'
+import { provider } from "web3-core"
+
+import { getTokenSupply } from "utils/setjsApi"
+import useWallet from "hooks/useWallet"
+import { bedTokenAddress, btc2xfliTokenAddress, dpiTokenAddress, eth2xfliTokenAddress, mviTokenAddress } from "constants/ethContractAddresses"
+import TokenSupplyContext from "./TokenSupplyContext"
+import BigNumber from "bignumber.js"
+
+const TokenSupplyProvider: React.FC = ({ children }) => {
+  const { ethereum }: { ethereum: provider } = useWallet()
+  const [dpiTotalSupply, setDpiTotalSupply] = useState<BigNumber>()
+  const [mviTotalSupply, setMviTotalSupply] = useState<BigNumber>()
+  const [bedTotalSupply, setBedTotalSupply] = useState<BigNumber>()
+  const [eth2xfliTotalSupply, setEth2xfliTotalSupply] = useState<BigNumber>()
+  const [btc2xfliTotalSupply, setBtc2xfliTotalSupply] = useState<BigNumber>()
+
+  useEffect(() => {
+    if (ethereum && dpiTokenAddress && mviTokenAddress && bedTokenAddress && eth2xfliTokenAddress && btc2xfliTokenAddress) {
+      getTokenSupply(ethereum, [dpiTokenAddress, mviTokenAddress, bedTokenAddress, eth2xfliTokenAddress, btc2xfliTokenAddress])
+        .then(result => {
+          const [dpiResult, mviResult, bedResult, eth2xFliResult, btc2xFliResult] = result
+          setDpiTotalSupply(new BigNumber(dpiResult.totalSupply.toString()).dividedBy(new BigNumber(10).pow(18)))
```

Comment:
even though Set returns an ethers `BigNumber`, index-ui uses `BigNumber` from bignumber.js so I elected to convert types here

Reply:
Makes sense! ethers' `BigNumber` can be a bit of a pain to use.

Reply:
Good to know. I've only use bignumber.js and Ethers BigNumber sparingly so I'm not sure which one is preferable for this repo to use. Seems safe to keep bignumber.js until we have convincing evidence to switch off

Reply:
Yeah, so basically ethers` BigNumber has a bunch of safety things built in (can't use floats) to ensure everything's compatible with the EVM (which doesn't allow floats). Way too onerous IMO though, I prefer converting to local BigNumber